poseidon-parameters/src/mds_matrix.rs

```diff
@@ -0,0 +1,147 @@
+use anyhow::Result;
+use ark_ff::{vec::Vec, PrimeField};
+
+use crate::{Matrix, MatrixOperations, SquareMatrix, SquareMatrixOperations};
+
+/// Represents an MDS (maximum distance separable) matrix.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct MdsMatrix<F: PrimeField>(pub SquareMatrix<F>);
+
+impl<F: PrimeField> MatrixOperations<F> for MdsMatrix<F> {
+    fn new(n_rows: usize, n_cols: usize, elements: Vec<F>) -> Self {
+        Self(SquareMatrix::new(n_rows, n_cols, elements))
+    }
+
+    fn elements(&self) -> &Vec<F> {
+        self.0.elements()
+    }
+
+    fn get_element(&self, i: usize, j: usize) -> F {
+        self.0.get_element(i, j)
+    }
+
+    fn set_element(&mut self, i: usize, j: usize, val: F) {
+        self.0.set_element(i, j, val)
+    }
+
+    fn rows(&self) -> Vec<&[F]> {
+        self.0.rows()
+    }
+
+    fn n_rows(&self) -> usize {
+        self.0.n_rows()
+    }
+
+    fn n_cols(&self) -> usize {
+        self.0.n_cols()
+    }
+
+    fn transpose(&self) -> Self {
+        Self(self.0.transpose())
+    }
+
+    fn hadamard_product(&self, rhs: &Self) -> Result<Self>
+    where
+        Self: Sized,
+    {
+        Ok(Self(self.0.hadamard_product(&rhs.0)?))
+    }
+}
+
+impl<F: PrimeField> MdsMatrix<F> {
+    /// Instantiate an MDS matrix from a list of elements.
+    ///
+    /// # Security
+    ///
+    /// You must ensure this matrix was generated securely,
+    /// using the Cauchy method in `fixed_cauchy_matrix` or
+    /// using the random subsampling method described in the original
+    /// paper.
+    pub fn from_elements(elements: Vec<F>) -> Self {
```

Comment:
Ah the functionality in `MdsMatrix::fixed_cauchy_matrix` is moved to a function in `poseidon-paramgen` instead of a method on `MdsMatrix` ✅